src/annotator/integrations/vitalsource.js

```diff
@@ -146,6 +146,38 @@ function getPDFPageImage() {
   );
 }
 
+/**
+ * Fix how a VitalSource book content frame scrolls, so that various related
+ * Hypothesis behaviors (the bucket bar, scrolling annotations into view) work
+ * as intended.
+ *
+ * Some VitalSource books (PDFs) make content scrolling work by making the
+ * content iframe really tall and having the parent frame scroll. This stops the
+ * Hypothesis bucket bar and scrolling annotations into view from working.
+ *
+ * @param {HTMLIFrameElement} frame
+ */
+function makeContentFrameScrollable(frame) {
+  if (frame.getAttribute('scrolling') !== 'no') {
+    // This is a book (eg. EPUB) where the workaround is not required.
+    return;
+  }
+
+  // Override inline styles of iframe (hence `!important`). The iframe lives
+  // in Shadow DOM, so the element styles won't affect the rest of the app.
+  const style = document.createElement('style');
+  style.textContent = `iframe { height: 100% !important; }`;
+  frame.insertAdjacentElement('beforebegin', style);
+
+  const removeScrollingAttr = () => frame.removeAttribute('scrolling');
+  removeScrollingAttr();
+
+  // Sometimes the attribute gets re-added by VS. Remove it if that
+  // happens.
```

Comment:
I believe the `<iframe>` is rendered as part of a component by a UI library/framework. I'm guessing that the re-addition of the scrolling attribute can happen if the component is re-rendered.